cmd/buildkitd/main_containerd_worker.go

```diff
@@ -355,6 +355,7 @@ func validContainerdSocket(cfg config.ContainerdConfig) bool {
 		return true
 	}
 	socketPath := strings.TrimPrefix(socket, "unix://")
+	socketPath = strings.TrimPrefix(socketPath, "npipe://")
```

Comment:
These lines should probably be moved into functions defined in platform specific files.